p2p/sentry/status_data_provider.go

```diff
@@ -76,6 +76,9 @@ func (s *StatusDataProvider) GetStatusData(ctx context.Context) (*proto_sentry.S
 
 func ReadChainHeadWithTx(tx kv.Tx) (ChainHead, error) {
 	header := rawdb.ReadCurrentHeader(tx)
+	if header == nil {
+		header = rawdb.ReadCurrentHeader2(tx)
```

Comment:
can we just call this directly instead of calling ReadCurrentHeader first? what matters to use is the which is the last block that has been sync-ed and get its header

Reply:
:+1:

Reply:
The only reason I originally did the patch like this was that I found the first call was failing and like Daniel I don't know  what the difference is between the data in the tables.

Once this is merged I'll overwrite my local fix before completing my PR.

Reply:
I've removed the ReadCurrentHeader call. One problem with it is that during initial sync (which can take a week) we will report to P2P that we are still on genesis, because HeadBlockKey is only updated at stage_finish. I think that conceptually it is correct (that we only report confirmed state blocks), but it is not nice for the P2P (because we are leeching for many days).

Reply:
Ideally we should report the last committed body. I'll make a follow-up with this.

Reply:
@mh0lt I will merge it now, because it fixes a production issue. I can help to rebase your PR and fix conflicts.